pkg/kinflate/transformers/namereferenceconfig.go

```diff
@@ -339,4 +339,109 @@ var defaultNameReferencePathConfigs = []referencePathConfig{
 			},
 		},
 	},
+	{
+		// StatefulSet references headless service, so need to update the references.
+		referencedGVK: schema.GroupVersionKind{
+			Version: "v1",
+			Kind:    "Service",
+		},
+		pathConfigs: []PathConfig{
+			{
+				GroupVersionKind: &schema.GroupVersionKind{
+					Group: "apps",
+					Kind:  "StatefulSet",
+				},
+				Path:               []string{"spec", "serviceName"},
+				CreateIfNotPresent: false,
+			},
+		},
+	},
+	{
+		referencedGVK: schema.GroupVersionKind{
+			Group: "rbac.authorization.k8s.io",
+			Kind:  "Role",
+		},
+		pathConfigs: []PathConfig{
+			{
+				GroupVersionKind: &schema.GroupVersionKind{
+					Group: "rbac.authorization.k8s.io",
+					Kind:  "RoleBinding",
+				},
+				Path:               []string{"roleRef", "name"},
+				CreateIfNotPresent: false,
+			},
+			{
+				GroupVersionKind: &schema.GroupVersionKind{
+					Group: "rbac.authorization.k8s.io",
+					Kind:  "ClusterRoleBinding",
```

Comment:
IIRC `ClusterRoleBinding` can't ref another `Role`.
I think we should drop this entry.